flask_appbuilder/security/views.py

```diff
@@ -735,11 +706,15 @@ def oauth_authorized(self, provider):
             except jwt.InvalidTokenError:
                 raise Exception("State signature is not valid!")
 
-            try:
-                next_url = state["next"][0] or self.appbuilder.get_url_for_index
-            except (KeyError, IndexError):
-                next_url = self.appbuilder.get_url_for_index
-
+            next_url = self.appbuilder.get_url_for_index
+            # Check if there is a next url on state
+            if "next" in state and len(state["next"]) > 0:
+                parsed_uri = urlparse(state["next"][0])
+                if parsed_uri.netloc != request.host:
```

Comment:
Is it possible to check the "base url" instead? I can't remember if that information makes it down to FAB or not (I suspect not?) but if it does checking that would be better.

Also should we check port number too? (which I guess could be optional/empty on either side?)

Reply:
Oh `request.host` already includes non-standard ports

Reply:
yes, so I think it's fine, do you agree?

Reply:
Yeah request.host is good.


Is it worth checking `parsed_uri.path` starts with https://flask.palletsprojects.com/en/2.0.x/api/?highlight=request#flask.Request.root_path for  too?